modules/tensor_mechanics/src/actions/TensorMechanicsAction.C

```diff
@@ -4,18 +4,33 @@
 /*          All contents are licensed under LGPL V2.1           */
 /*             See LICENSE for full restrictions                */
 /****************************************************************/
-#include "TensorMechanicsAction.h"
-#include "Factory.h"
-#include "FEProblem.h"
 #include "Conversion.h"
 #include "FEProblem.h"
 #include "Factory.h"
 #include "MooseMesh.h"
 #include "MooseObjectAction.h"
+#include "TensorMechanicsAction.h"
 
 #include "libmesh/string_to_enum.h"
 #include <algorithm>
 
+const std::map<std::string, std::string> TensorMechanicsAction::_ranktwoaux_table = {
+    {"strain", "total_strain"},
+    {"stress", "stress"},
+    {"elastic_strain", "elastic_strain"},
+    {"plastic_strain", "plastic_strain"},
+    {"creep_strain", "creep_strain"}};
+const std::vector<char> TensorMechanicsAction::_component_table = {'x', 'y', 'z'};
+const std::map<std::string, std::string> TensorMechanicsAction::_ranktwoscalaraux_table = {
+    {"vonmises", "VonMisesStress"},
+    {"hydrostatic", "Hydrostatic"},
+    {"stress_max", "MaxPrincipal"},
+    {"stress_mid", "MidPrincipal"},
+    {"stress_min", "MinPrincipal"},
+    {"firstinv", "FirstInvariant"},
```

Comment:
I was going with the de facto standard in the tests. If I change this I'll have to regold as well. I can do that course. That would allow me to make the code support invariants of the strain more elegantly add well.